src/dc_mimeparser.rs

```diff
@@ -133,7 +132,15 @@ pub unsafe fn dc_mimeparser_parse<'a>(context: &'a Context, body: &[u8]) -> dc_m
         dc_mimeparser_parse_mime_recursive(mimeparser_ref, mimeparser_ref.mimeroot);
         let field: *mut mailimf_field = dc_mimeparser_lookup_field(&mimeparser, "Subject");
         if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_SUBJECT as libc::c_int {
-            mimeparser.subject = dc_decode_header_words((*(*field).fld_data.fld_subject).sbj_value)
+            let decoded = dc_decode_header_words((*(*field).fld_data.fld_subject).sbj_value);
+            if decoded.is_null()
+            /* XXX: can it happen? */
```

Comment:
Need to replace with assert if can't

Reply:
From reading code of `dc_decode_header_words`, it can return NULL, in principle. Can it return for inputs, that can happen in this function, I do not know.